src/js/components/Carousel.js

```diff
@@ -58,6 +70,33 @@ export default class Carousel extends Component {
     scrollParents.forEach(function (scrollParent) {
       scrollParent.removeEventListener('scroll', this._handleScroll);
     }.bind(this));
+
+    this._unmountHammer();
+  }
+
+  _unmountHammer () {
+    if (this.hammer) {
```

Comment:
do you why you did not have to add `_unmountHammer` and `_updateHammer ` to Carousel constructor and change the bind context?

Shouldn't we be adding something like this:

```
this._unmountHammer = this._unmountHammer(this);
this._updateHammer = this._updateHammer(this);
```